pkg/virt-handler/isolation/process_test.go

```diff
@@ -0,0 +1,119 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package isolation
+
+import (
+	"github.com/mitchellh/go-ps"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("process", func() {
+	const (
+		processTestExecPath = "processA"
+		processTestPID      = 110
+		nonExistPPid        = 300
+	)
+	emptyProcessList := []ProcessStub{}
+	procStub1 := ProcessStub{ppid: 1, pid: 120, binary: processTestExecPath}
+	procStub2 := ProcessStub{ppid: processTestPID, pid: 2222, binary: "processB"}
+	procStub3 := ProcessStub{ppid: 1, pid: 110, binary: "processC"}
+	procStub4 := ProcessStub{ppid: processTestPID, pid: 3333, binary: "processD"}
+	testProcesses := []ProcessStub{procStub1, procStub3, procStub2, procStub4}
+
+	Context("find child processes", func() {
+		table.DescribeTable("should return the correct child processes of the given pid",
+			func(processes []ProcessStub, ppid int, expectedProcesses []ProcessStub) {
+				Expect(childProcesses(convertToPsGoProcessList(processes), ppid)).
+					To(ConsistOf(convertToPsGoProcessList(expectedProcesses)))
+			},
+			table.Entry("given no input processes, there are no child processes",
+				emptyProcessList, nonExistPPid, emptyProcessList,
+			),
+			table.Entry("given process list and non-exist pid, should return no child processes",
+				testProcesses, nonExistPPid, emptyProcessList,
+			),
+			table.Entry("given process list and pid where there are child processes of the given pid",
+				testProcesses, processTestPID, []ProcessStub{procStub2, procStub4},
```

Comment:
Here you want `[]ps.Process{procStub2, procStub4}` not this stub array.

Reply:
Done